From: LibreELEC
Subject: [PATCH] include climits where the limit macros are used

Several sources use INT_MAX, INT_MIN and LLONG_MAX without including
<climits>, relying on it arriving through another header. It does on the
host toolchain but not on the cross toolchain, where they fail to compile.

--- a/goo/GooString.cc
+++ b/goo/GooString.cc
@@ -41,6 +41,7 @@
 
 #include <config.h>
 
+#include <climits>
 #include <cassert>
 #include <cctype>
 #include <cmath>
--- a/poppler/Form.cc
+++ b/poppler/Form.cc
@@ -49,6 +49,7 @@
 #include <array>
 #include <set>
 #include <limits>
+#include <climits>
 #include <cstddef>
 #include <cstdlib>
 #include <cstring>
--- a/poppler/Function.cc
+++ b/poppler/Function.cc
@@ -30,6 +30,7 @@
 
 #include <config.h>
 
+#include <climits>
 #include <cstdlib>
 #include <cstring>
 #include <cctype>
--- a/poppler/Gfx.cc
+++ b/poppler/Gfx.cc
@@ -60,6 +60,7 @@
 
 #include <config.h>
 
+#include <climits>
 #include <cstdlib>
 #include <cstdio>
 #include <cstring>
--- a/poppler/Parser.cc
+++ b/poppler/Parser.cc
@@ -33,6 +33,8 @@
 
 #include <config.h>
 
+#include <climits>
+
 #include "Object.h"
 #include "Array.h"
 #include "Dict.h"
--- a/poppler/SplashOutputDev.cc
+++ b/poppler/SplashOutputDev.cc
@@ -53,6 +53,7 @@
 
 #include <config.h>
 
+#include <climits>
 #include <cstring>
 #include <cmath>
 #include <vector>
--- a/poppler/TextOutputDev.cc
+++ b/poppler/TextOutputDev.cc
@@ -56,6 +56,7 @@
 
 #include <config.h>
 
+#include <climits>
 #include <cstdio>
 #include <cstdlib>
 #include <cstddef>
